examples/gamepad.html

```diff
@@ -92,68 +124,95 @@
-        var changed = false;
+        var gamepads = navigator.getGamepads ? navigator.getGamepads() :
+          (navigator.webkitGetGamepads ? navigator.webkitGetGamepads()
+            : []);
         for (var i = 0; i < gamepads.length; ++i) {
           var pad = gamepads[i];
           if (pad != null) {
-            if (roll != pad.axes[2] || pitch != pad.axes[1] || yaw != pad.axes[0]) {
-              changed = true;
-              yaw = pad.axes[0];
-              pitch = pad.axes[1];
-              roll = pad.axes[2];
-              textRotation.innerHTML = "yaw " + Math.round(yaw * 180) 
-                + " pitch " + Math.round(pitch * 180) 
-                + " roll " + Math.round(roll * 180);
+            if (!gamepad) {
+              gamepad=true;
+              divRotator.style.display = 'none';
+            }
+            if (axes[0] != pad.axes[0] || axes[1] != pad.axes[1]
+                || axes[2]  != pad.axes[2]) {
+              axes[0] = pad.axes[0];
+              axes[1] = pad.axes[1];
+              axes[2] = pad.axes[2];
+              updateDirection(axes[0] * Math.PI,
+                             axes[1] * Math.PI,
+                             axes[2] * Math.PI);
             }
-            break;
+            window.requestAnimationFrame(runAnimation);
+            return;
           }
         }
-
-        if (changed) {
-          // from http://mathworld.wolfram.com/EulerAngles.html
-          // down is back
-          
-          var phi = yaw * Math.PI; 
-          // 1. the first rotation is by an angle phi about the z-axis using D,
-          
-          var theta = pitch * Math.PI; 
-          // the second rotation is by an angle theta in [0,pi] about the former x-axis (now x^') using C, and
-          
-          var psi = roll * Math.PI;   
-          //3. the third rotation is by an angle psi about the former z-axis (now z^') using B.
-
-          rotator.setRotationMatrix([
-            Math.cos(psi) * Math.cos(phi) - Math.cos(theta) * Math.sin(phi) * Math.sin(psi),
-            Math.cos(psi) * Math.sin(phi) + Math.cos(theta) * Math.cos(phi) * Math.sin(psi),
-            Math.sin(psi) * Math.sin(theta),
-            -Math.sin(psi) * Math.cos(phi) - Math.cos(theta) * Math.sin(phi) * Math.cos(psi),
-            -Math.sin(psi) * Math.sin(phi) + Math.cos(theta) * Math.cos(phi) * Math.cos(psi),
-            Math.cos(psi) * Math.sin(theta),
-            Math.sin(theta) * Math.sin(phi),
-            -Math.sin(theta) * Math.cos(phi),
-            Math.cos(theta)
-          ]);
-          decoder.setRotationMatrix(rotator.getRotationMatrix());
+        if (gamepad) {
+          divRotator.style.display = 'block';
+          selectRotation(0);
+          gamepad = false;
         }
-        window.requestAnimationFrame(runAnimation);
+        setTimeout(runAnimation, 500);
```

Comment:
It is just to save CPU time. If no gamepad is present, then you do not have to update that frequent.